### ISSUE TITLE
Exception while invoking method 'dumbCollectionGetNew' Error: Match error: Failed Match

### ISSUE THREAD

**Issue by jamgold** [richsilv/meteor-dumb-collections#3]
Testing my site (Meteor 1.0.2.1) on different browsers I suddenly hit this on Google Chrome, no idea why
```
Exception while invoking method 'dumbCollectionGetNew' Error: Match error: Failed Match.OneOf or Match.Optional validation
at checkSubtree (packages/check/match.js:244:1)
at check (packages/check/match.js:32:1)
at [object Object]._.extend._getFindOptions (packages/mongo/collection.js:238:1)
at [object Object]._.extend.find (packages/mongo/collection.js:274:1)
at [object Object].Meteor.methods.dumbCollectionGetNew (packages/richsilv:dumb-collections/richsilv:dumb-collections.js:25:1)
at maybeAuditArgumentChecks (packages/ddp/livedata_server.js:1599:1)
at packages/ddp/livedata_server.js:648:1
at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
at packages/ddp/livedata_server.js:647:1
at [object Object]._.extend.withValue (packages/meteor/dynamics_nodejs.js:56:1)
Sanitized and reported to the client as: Match failed [400]
```

I am running version 1.1.2 of dumb-collections

**Comment by richsilv** [comment]
Thanks for raising. Can you list the other packages you're using?

**Comment by jamgold** [comment]
Eeek. A lot of modules. I will try to recreate a repro
```
accounts-facebook                  1.0.3  Login service for Facebook accounts
accounts-google                    1.0.3  Login service for Google accounts
accounts-password                  1.0.5  Password support for accounts
alanning:roles                     1.2.13  Role-based authorization
aldeed:autoform                    4.2.1* Easily create forms with automatic insert and update, and automatic reactive validation.
aldeed:collection2                 2.3.0  Automatic validation of insert and update operations on the client and server.
aldeed:simple-schema               1.2.0* A simple schema validation object with reactivity. Used by collection2 and autoform.
benjaminrh:event-hooks             1.4.3  Provides hooks for various user-triggered events
email                              1.0.5  Send email messages
iron:router                        1.0.6  Routing specifically designed for Meteor
jquery                             1.0.2  Manipulate the DOM using CSS selectors
lepozepo:s3                        4.1.3  Upload files to S3. Allows use of Knox Server-Side.
mrt:accounts-admin-ui-bootstrap-3  0.2.7  A roles based account management system using bootstrap 3
mrt:accounts-ui-bootstrap-3        0.3.3  Bootstrap styled version of login widgets
mrt:googlemaps                     0.0.2  google maps repackaged for Meteor
mrt:modernizr-meteor               2.6.2  Modernizr repackaged for Meteor
natestrauser:font-awesome          4.2.0* Latest version Font-Awesome loaded via CDN
percolate:velocityjs               1.1.0  Wrapper for velocity.js includes ui pack
richsilv:dumb-collections          1.1.2  Meteor Collections which only sync with the server on request and are saved in local storage.
sergeyt:typeahead                  0.10.5_7+ LOCAL Autocomplete package for meteor powered by twitter typeahead.js
spacebars                          1.0.4  Handlebars-like template language for Meteor
standard-app-packages              1.0.4  Moved to meteor-platform
underscore                         1.0.2  Collection of small helpers: _.map, _.each, ...
```

**Comment by jamgold** [comment]
OK, just created a brand new meteor project with only richsilv:dumb-collections, and the same thing happens with Safari, Chrome and Firefox

**Comment by richsilv** [comment]
Thanks, that should make things more straightforward. I will look at this
tomorrow AM (London) and revert.

**Comment by jamgold** [comment]
Cool. Here is what I found. The method
```
dumbCollectionGetNew: function(existing, name, query, options)
```
is being called with options = null by default, which screws up the collections[name].find()

Simply changing it to
```
return collections[name].find(_.extend(query || {}, {
				_id: {
					$nin: existing
				}
			}), options || {}).fetch()
```
fixed it

**Comment by richsilv** [comment]
Spot on, thanks.  This is obviously a new validation added to the `find` method in the latest version, as passing an undefined object worked okay before.  Fixed with a8343399c824fb3b333b71cafae3fc2c9c305af1.
